### PR TITLE
Fix libtiff CVEs: 2022-2056, 2022-2057, 2022-2058

### PR DESCRIPTION
**Pull request by mandeepsplaha**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Patch CVE-2022-2056, CVE-2022-2057, CVE-2022-2058 in libtiff package. Patch taken from upstream.

###### Change Log  

- CVE-2022-2056
- CVE-2022-2057
- CVE-2022-2058

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-2056
- https://nvd.nist.gov/vuln/detail/CVE-2022-2057
- https://nvd.nist.gov/vuln/detail/CVE-2022-2058

###### Test Methodology

- Buddy build: [213950](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=213950&view=results)

### CODE REVIEW

File: SPECS/libtiff/libtiff.spec [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 Version:        4.4.0
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        libtiff
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Libraries
 URL:            https://gitlab.com/libtiff/libtiff
 Source0:        https://gitlab.com/libtiff/libtiff/-/archive/v%{version}/libtiff-v%{version}.tar.gz
+Patch0:         CVE-2022-2056_CVE-2022-2057_CVE-2022-2058.patch
```
Comment (jslobodzian) [inline SPECS/libtiff/libtiff.spec:11]: I don't think that will work correctly.
It should be a single patch file per patch.

Comment (jslobodzian) [reply SPECS/libtiff/libtiff.spec:11]: It's more about the scan tooling than the change

Comment (jslobodzian) [reply SPECS/libtiff/libtiff.spec:11]: We can check with Nick.

Alternate option is to similar to what we discussed yesterday.  Create one patch file and then consider 3 indpendent nopatch files.

Comment (mandeepsplaha) [reply SPECS/libtiff/libtiff.spec:11]: Got it. I can make that change.

Comment (PawelWMS) [inline SPECS/libtiff/libtiff.spec:11]: We usually do one CVE fix per line. I think our tooling might depend on the correct format of the names of the patch files.

File: SPECS/libtiff/libtiff.spec [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 Version:        4.4.0
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        libtiff
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Libraries
 URL:            https://gitlab.com/libtiff/libtiff
 Source0:        https://gitlab.com/libtiff/libtiff/-/archive/v%{version}/libtiff-v%{version}.tar.gz
+Patch0:         CVE-2022-2056.patch
```
Comment (PawelWMS) [inline SPECS/libtiff/libtiff.spec:11]: I'd also mention the fix in a comment here:
```suggestion
# Also fixes CVE-2022-2057 and CVE-2022-2058.
Patch0:         CVE-2022-2056.patch
```

**Review by PawelWMS — APPROVED** [review]
One more suggestion but looks good.

**Pull request merged**